src/mongocrypt-marking.c

```diff
@@ -1118,6 +1404,10 @@ _mongocrypt_marking_to_ciphertext (void *ctx,
       } else {
          BSON_ASSERT (marking->fle2.type ==
                       MONGOCRYPT_FLE2_PLACEHOLDER_TYPE_FIND);
+         if (marking->fle2.algorithm == MONGOCRYPT_FLE2_ALGORITHM_RANGE) {
```

Comment:
Suggest converting to `switch` statement on permitted values of `mongocrypt_fle2_encryption_algorithm_t`.